src/os/bluestore/BlueStore.cc

```diff
@@ -8437,6 +8437,7 @@ int BlueStore::_collection_list(
     }
   }
   dout(20) << __func__ << " pend " << pretty_binary_string(pend) << dendl;
+  ls->reserve(max);
```

Comment:
```
 ceph version 14.0.1-771-g680ef7c (680ef7c847bac83761172d65ca75edda38014eba) nautilus (dev)
 1: (()+0xf5d0) [0x7f22274d85d0]
 2: (gsignal()+0x37) [0x7f22262cf207]
 3: (abort()+0x148) [0x7f22262d08f8]
 4: (__gnu_cxx::__verbose_terminate_handler()+0x165) [0x7f2226bde7d5]
 5: (()+0x5e746) [0x7f2226bdc746]
 6: (()+0x5e773) [0x7f2226bdc773]
 7: (()+0x5e993) [0x7f2226bdc993]
 8: (()+0x1772b) [0x7f2229e2a72b]
 9: (tcmalloc::allocate_full_cpp_throw_oom(unsigned long)+0xf3) [0x7f2229e48a03]
 10: (HashIndex::list_by_hash_bitwise(std::vector<std::string, std::allocator<std::string> > const&, ghobject_t const&, int, ghobject_t*, std::
vector<ghobject_t, std::allocator<ghobject_t> >*)+0x853) [0x55af0d626f53]
 11: (HashIndex::_collection_list_partial(ghobject_t const&, ghobject_t const&, int, std::vector<ghobject_t, std::allocator<ghobject_t> >*, gho
bject_t*)+0x12b) [0x55af0d62753b]
 12: (FileStore::collection_list(coll_t const&, ghobject_t const&, ghobject_t const&, int, std::vector<ghobject_t, std::allocator<ghobject_t> >
*, ghobject_t*)+0x259) [0x55af0d443c59]
 13: (PGBackend::objects_list_range(hobject_t const&, hobject_t const&, std::vector<hobject_t, std::allocator<hobject_t> >*, std::vector<ghobje
ct_t, std::allocator<ghobject_t> >*)+0x147) [0x55af0d2d50e7]
 14: (PG::build_scrub_map_chunk(ScrubMap&, ScrubMapBuilder&, hobject_t, hobject_t, bool, ThreadPool::TPHandle&)+0x28a) [0x55af0d18812a]
 15: (PG::chunky_scrub(ThreadPool::TPHandle&)+0x15bd) [0x55af0d1b035d]
 16: (PG::scrub(unsigned int, ThreadPool::TPHandle&)+0xaf) [0x55af0d1b143f]
 17: (PGScrub::run(OSD*, OSDShard*, boost::intrusive_ptr<PG>&, ThreadPool::TPHandle&)+0x12) [0x55af0d35d6b2]
 18: (OSD::ShardedOpWQ::_process(unsigned int, ceph::heartbeat_handle_d*)+0xb6c) [0x55af0d0e9dbc]
```

this breaks tcmalloc. 

see

https://github.com/ceph/ceph/blob/0aa5566c81b09175971644f9d85e665919b8abef/src/osd/PGBackend.cc#L388-L394

we cannot `reserve(max)` . sometimes `max` is really large.

Reply:
i forget this INT_MAX.  i modified some places which used INT_MAX in tests. I think using INT_MAX is  a bug which used wrong parameters. If  collection has many objects, it still allocate memory.
ObjectStore provois this data:https://github.com/ceph/ceph/blob/0aa5566c81b09175971644f9d85e665919b8abef/src/os/ObjectStore.h#L1597.  So we can use this.